api/src/services/uploads/uploads.ts

```diff
@@ -147,116 +154,192 @@ export const Upload: UploadRelationResolvers = {
-{
-  '1A': {
-    '1': {
-      id: 1,
-      filename: 'file1.csv',
-      objectKey: 'uploads/1/1/1/1/file1.csv',
-      ...
-    },
-    '2': {
-      id: 2,
-      filename: 'file2.csv',
-      objectKey: 'uploads/1/2/1/2/file2.csv',
-      ...
-    },
-    ...
-  },
-  '1B': {
-    '1': {
-      id: 3,
-      filename: 'file3.csv',
-      objectKey: 'uploads/1/1/1/3/file3.csv',
-      ...
-    },
-    ...
-  },
-  ...
+type UserObj = {
+  email: string
+  id: number
 }
+type UploadPayload = {
+  objectKey: string
+  filename: string
+  createdAt: Date
+}
+type UploadInfoForProject = {
+  organization: OrganizationObj
+  user: UserObj
+  outputTemplateId: number
+  ProjectType: string
+  uploadsToAdd: Partial<Record<AgencyId, UploadPayload>>
+  uploadsToRemove: Partial<Record<AgencyId, UploadPayload>>
+}
+type InfoForSubrecipient = {
+  organization: OrganizationObj
+  user: UserObj
+  outputTemplateId: number
+}
+type InfoForArchive = {
+  organization: OrganizationObj
+}
+/*
+This type should be similar to the following python class:
+class ProjectLambdaPayload(BaseModel):
 */
-export const getUploadsByExpenditureCategory = async () => {
-  const organization = await db.organization.findFirst({
-    where: { id: context.currentUser.agency.organizationId },
-  })
+export type ProjectLambdaPayload = Record<
+  ExpenditureCategoryCode,
+  UploadInfoForProject
+>
+export type SubrecipientLambdaPayload = Record<
+  'Subrecipient',
+  InfoForSubrecipient
+>
+export type CreateArchiveLambdaPayload = Record<'zip', InfoForArchive>
 
-  const reportingPeriod = await db.reportingPeriod.findFirst({
-    where: { id: organization.preferences['current_reporting_period_id'] },
-  })
+export const getUploadsByExpenditureCategory = async (
+  organization: Organization,
+  reportingPeriod: ReportingPeriod
+): Promise<ProjectLambdaPayload> => {
+  const validUploadsInPeriod: UploadsWithValidationsAndExpenditureCategory[] =
+    await getValidUploadsInCurrentPeriod(organization, reportingPeriod)
 
-  // valid upload validations
-  const validUploadIds = await db.uploadValidation.findMany({
-    where: {
-      passed: true,
-      upload: { agency: { organizationId: organization.id } },
-    },
-    select: {
-      uploadId: true,
-    },
-    distinct: ['uploadId'],
-  })
+  const uploadsByEC: ProjectLambdaPayload = {}
 
-  const uploadsForPeriod = await db.upload.findMany({
+  // Get the most recent upload for each expenditure category and agency and set the S3 Object key
+  for (const upload of validUploadsInPeriod) {
+    const uploadPayload: UploadPayload = {
+      objectKey: await getS3UploadFileKey(organization.id, upload),
+      createdAt: upload.createdAt,
+      filename: upload.filename,
+    }
+
+    if (!uploadsByEC[upload.expenditureCategory.code]) {
+      // The EC code was never added. This is the time to initialize it.
+      uploadsByEC[upload.expenditureCategory.code] = {
+        organization: {
+          id: organization.id,
```

Comment:
Nit but any reason not to just do `organization` here?

Reply:
the `organization` here is of type `Organization` which contains many other fields like `name` the usual timestamps, etc so figured I'll pair it down to only the relevant fields to send to the step-function.

Although this brings up a good point- maybe when I query for the organization [here](https://github.com/usdigitalresponse/cpf-reporter/pull/395/files#diff-f45033b7d2aebc409368859184ed65ef5836a301a645062940ad896a6e1b2295R348) I can only select these specific fields that way I don't have to re-define this everywhere.